File: ios/cross_bluetooth_api/Sources/cross_bluetooth_api/Events/Event.swift
import Foundation
import Flutter

enum EventName: String, Codable {
    case gattserverdisconnected
    case characteristicvaluechanged
}

public class Event: NSObject {
    let name: EventName
    
    init(name: EventName) {
        self.name = name
    }

    func toMap() -> [String:AnyObject] {
        return [
            "name": name.rawValue
        ] as [String:AnyObject]
    }
}

File: android/src/main/kotlin/io/github/kokemus/cross_bluetooth_api/events/Event.kt
package io.github.kokemus.cross_bluetooth_api.events

enum class EventName(val rawValue: String) {
    GATT_SERVER_DISCONNECTED("gattserverdisconnected"),
    CHARACTERISTIC_VALUE_CHANGED("characteristicvaluechanged")
}

open class Event(private val name: EventName) {
    open fun toMap(): Map<String, Any> {
        return mapOf("name" to name.rawValue)
    }
}
